prime-dotnet-webapi/Services/HealthAuthorityService.cs

```diff
@@ -148,17 +154,23 @@ public async Task UpdatePrivacyOfficeAsync(int healthAuthorityId, PrivacyOfficeV
 
         public async Task UpdateVendorsAsync(int healthAuthorityId, IEnumerable<int> vendorCodes)
         {
-            var oldVendors = await _context.HealthAuthorityVendors
+            var existingVendors = await _context.HealthAuthorityVendors
                 .Where(ct => ct.HealthAuthorityOrganizationId == healthAuthorityId)
                 .ToListAsync();
 
-            _context.HealthAuthorityVendors.RemoveRange(oldVendors);
+            var deletedVendors = existingVendors
+                .Except(existingVendors
+                    .Where(ev => vendorCodes.Contains(ev.VendorCode)));
 
-            var newVendors = vendorCodes.Select(code => new HealthAuthorityVendor
-            {
-                HealthAuthorityOrganizationId = healthAuthorityId,
-                VendorCode = code
-            });
+            _context.HealthAuthorityVendors.RemoveRange(deletedVendors);
+
+            var newVendors = vendorCodes
+                .Except(existingVendors.Select(ev => ev.VendorCode))
+                .Select(code => new HealthAuthorityVendor
+                {
+                    HealthAuthorityOrganizationId = healthAuthorityId,
+                    VendorCode = code
+                });
```

Comment:
I will show you how to use the `EntityMatcher` to greatly simplify this code

Reply:
Sounds good, that code existed from before though.

Reply:
Refactored